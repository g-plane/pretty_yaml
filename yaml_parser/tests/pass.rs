use insta::{assert_snapshot, glob, Settings};
use std::{fs, path::Path};

#[test]
fn pass_snapshot() {
    glob!("pass/*.yaml", |path| {
        let input = fs::read_to_string(path).unwrap();
        let tree = match yaml_parser::parse(&input) {
            Ok(tree) => tree,
            Err(err) => panic!("failed to parse '{}': {err}", path.display()),
        };
        assert_eq!(
            tree.to_string(),
            input,
            "syntax tree of '{}' does not match source",
            path.display()
        );

        build_settings(path).bind(|| {
            let name = path.file_stem().unwrap().to_str().unwrap();
            assert_snapshot!(name, format!("{tree:#?}"));
        });
    });
}

fn build_settings(path: &Path) -> Settings {
    let mut settings = Settings::clone_current();
    settings.set_snapshot_path(path.parent().unwrap());
    settings.remove_snapshot_suffix();
    settings.set_prepend_module_to_snapshot(false);
    settings.remove_input_file();
    settings.set_omit_expression(true);
    settings.remove_input_file();
    settings.remove_info();
    settings
}
