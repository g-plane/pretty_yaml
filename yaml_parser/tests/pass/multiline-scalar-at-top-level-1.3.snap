---
source: yaml_parser/tests/pass.rs
---
ROOT@0..19
  DOCUMENT@0..5
    DIRECTIVES_END@0..3 "---"
    WHITESPACE@3..4 "\n"
    FLOW@4..5
      PLAIN_SCALAR@4..5 "a"
  WHITESPACE@5..6 "\n"
  DOCUMENT@6..13
    FLOW@6..13
      PLAIN_SCALAR@6..13 "b  \n  c"
  WHITESPACE@13..14 "\n"
  DOCUMENT@14..15
    FLOW@14..15
      PLAIN_SCALAR@14..15 "d"
  WHITESPACE@15..17 "\n\n"
  DOCUMENT@17..18
    FLOW@17..18
      PLAIN_SCALAR@17..18 "e"
  WHITESPACE@18..19 "\n"
